Add expressions.html debug page with barebones expression editor

anandthakker opened the pull request:
Run `yarn start-debug` and open http://localhost:9966/debug/expressions.html to view the page.

anandthakker commented:
FYI @nickidlugash @aparlato  -  this debug page will let you see how old-style stop functions get converted to expressions under the hood

mollymerp commented on src/style-spec/function/index.js:
```diff
@@ -90,6 +90,9 @@ function createFunction(parameters: FunctionParameters, propertySpec: StylePrope
         }
+
+        // useful for debugging, especially for converted stop functions
+        f.rawExpression = expr;
```
should we conditionally save this property (i.e. add a debug argument/option to `createFunction`)? not sure it would make a big difference beyond clarity on what `rawExpression` is used for.

anandthakker replied:
Yeah, I considered that briefly, but AFAIK the current build process doesn't have an easy way to do `if (DEBUG BUILD) {...}`.  We could use `envify`, which we're already using for the benchmarks, but since this seems pretty harmless, I thought maybe it wasn't worth adding a whole transform to the browserify build...

mollymerp replied:
yeah let's go with not over-complicating things 👌

mollymerp commented on src/style-spec/function/index.js:
```diff
@@ -90,6 +90,9 @@ function createFunction(parameters: FunctionParameters, propertySpec: StylePrope
+        f.rawExpression = expr;
         return f;
     } else {
         console.log(JSON.stringify(expr, null, 2));
```
same as above perhaps 💭

anandthakker replied:
Oops, yeah this is leftover, will remove

anandthakker commented:
> Also checking that you meant to keep the changes to circles.html from your presentation this morning?

ya, I figured we might as well

anandthakker commented on debug/expression_editor.js:
```diff
@@ -0,0 +1,147 @@
+const util = require('../src/util/util');
+const spec = require('../src/style-spec/reference/v8.json');
+
+class ExpressionEditor extends React.Component {
```
Good point.  You _can_ edit expressions here -- it'll accept constants, old-style functions, _or_ new-style expressions... so maybe `StylePropertyEditor` would be a more appropriate name

anandthakker commented:
@mollymerp addressed your comments ( 🙇 ) -- going to merge now, but also setting a reminder for 3 months from now to remove this: it's useful right now because expressions are new (and with a few parts of the spec still shifting slightly), but once they're "mainstream", this will probably be stale / clutter up the debug directory
